operator/v1/types_ingress.go

```diff
@@ -282,6 +287,230 @@ var (
+	// DH key for a 2048 bits RSA key), but will not exceed this maximum value. Only 1024 or 2048
+	// values are allowed.
+	//
+	// If unset, the "Intermediate" DH Parameter size [1] is used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	DHParamSize *DHParamSize `json:"dhParamSize,omitempty"`
+}
+
+// SecurityProtocol defines one or more security protocols used by
+// an IngressController to secure network connections.
+type SecurityProtocol struct {
+	// minimumVersion enforces use of SecurityProtocolVersion or newer on
+	// SSL connections initiated by an IngressController. minimumVersion
+	// must be lower than maximumVersion.
+	//
+	// If unset and maximumVersion is set, minimumVersion will be set
+	// to maximumVersion. If minimumVersion and maximumVersion are unset,
+	// the minimum version in "Intermediate" Versions [1] is used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	MinimumVersion *SecurityProtocolVersion `json:"minimumVersion,omitempty"`
+	// maximumVersion enforces use of SecurityProtocolVersion or older on
+	// SSL connections initiated by an IngressController. maximumVersion
+	// must be higher than minimumVersion.
+	//
+	// If unset and minimumVersion is set, maximumVersion will be set
+	// to minimumVersion. If minimumVersion and maximumVersion are unset,
+	// the maximum version in "Intermediate" Versions [1] is used:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28default.29
+	//
+	// +optional
+	MaximumVersion *SecurityProtocolVersion `json:"maximumVersion,omitempty"`
+}
+
+// SecurityProtocolVersion is a way to specify the TLS security protocol version
+// used for securing IngressController network connections.
+type SecurityProtocolVersion string
+
+const (
+	// TLSv1.0 is version 1.0 of the TLS security protocol used for securing
+	// IngressController network connections.
+	SecurityProtocolTLS10Version SecurityProtocolVersion = "TLSv1.0"
+	// TLSv1.1 is version 1.1 of the TLS security protocol used for securing
+	// IngressController network connections.
+	SecurityProtocolTLS11Version SecurityProtocolVersion = "TLSv1.1"
+	// TLSv1.2 is version 1.2 of the TLS security protocol used for securing
+	// IngressController network connections.
+	SecurityProtocolTLS12Version SecurityProtocolVersion = "TLSv1.2"
+	// TLSv1.3 is version 1.3 of the TLS security protocol used for securing
+	// IngressController network connections.
+	SecurityProtocolTLS13Version SecurityProtocolVersion = "TLSv1.3"
+)
+
+// DHParamSize sets the maximum size of the Diffie-Hellman parameters used for
+// generating the ephemeral/temporary Diffie-Hellman key.
+type DHParamSize string
+
+const (
+	// 1024 is a Diffie-Hellman parameter of 1024 bits.
+	DHParamSize1024 DHParamSize = "1024"
+	// 2048 is a Diffie-Hellman parameter of 2048 bits.
```

Comment:
I think you want to start with the field name, not the value: `1024 is a Diffie-Hellman parameter of 1024 bits.`→`DHParamSize1024 is a Diffie-Hellman parameter of 1024 bits.` and `2048 is a Diffie-Hellman parameter of 2048 bits.`→`DHParamSize2048 is a Diffie-Hellman parameter of 2048 bits.`.

Reply:
@Miciah in https://github.com/openshift/api/pull/350#discussion_r294471202 I was asked by @ironcladlou to use constant values instead of names. I see values being used for godocs for all other constants of the `IngressController` api type.